sympy/stats/rv.py

```diff
@@ -1265,19 +1286,21 @@ def given_fn_subs(*args):
     else:
         fn = lambdify(rvs, expr, modules=library, **kwargs)
 
+
     if condition is not None:
         given_fn = lambdify(rvs, condition, **kwargs)
 
     def return_generator_infinite():
         count = 0
         np = import_module('numpy')
         if np:
-            rand_state = np.random.default_rng(seed=seed)
+            rand_state = np.random.default_rng(seed=seed) if library != "pymc3" else seed
```

Comment:
Ideally we wouldn't special case a particular library here. Can the different classes not just handle this differently?

Is this what you meant about changes in a followup PR?

Reply:
This will be removed and handled in the different classes in the next commit.